### PR TITLE
APPSRE-7775 bump nodejs to next lts

### PR DESCRIPTION
**Pull request by fishi0x01**
nodejs16 reached EOL. Lets bump to next LTS 18.

Note: starting with node17 there is funny stuff with openssl. For now, setting `--openssl-legacy-provider` for compatibility. However, long-term we should consider upgrading `react-scripts`.

### CODE REVIEW

**Comment by rporres** [comment]
Can you provide documentation/details about the "funny stuff"? A link may suffice.

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "copy-img": "ncp node_modules/patternfly/dist/img src/img",
     "build-css": "sass --style compressed $npm_package_sassIncludes_src $npm_package_sassIncludes_patternfly $npm_package_sassIncludes_patternflyReact $npm_package_sassIncludes_bootstrap $npm_package_sassIncludes_fontAwesome src/",
     "watch-css": "yarn build-css && sass --watch --style compressed $npm_package_sassIncludes_src $npm_package_sassIncludes_patternfly $npm_package_sassIncludes_patternflyReact $npm_package_sassIncludes_bootstrap $npm_package_sassIncludes_fontAwesome src/",
-    "build-js": "react-scripts build",
+    "build-js": "react-scripts --openssl-legacy-provider build",
```
Comment (hemslo) [inline package.json:50]: other `react-scripts` also need `--openssl-legacy-provider`, like `start:local`

**Review by hemslo — COMMENTED** [review]
`--openssl-legacy-provider` details can be found https://github.com/facebook/create-react-app/issues/11562. Can cleanup after we upgrade dependencies.

File: package.json (hunk 1 above)
Comment (fishi0x01) [reply package.json:50]: Added flag to other `react-scripts` targets too now.

**Review by hemslo — APPROVED** [review]
LGTM!

File: package.json (hunk 1 above)
Comment (fishi0x01) [reply package.json:50]: @hemslo it seems the `--openssl-legacy-provider` option is not allowed in every `react-scripts` context. 

```
yarn lint && react-scripts --openssl-legacy-provider test --env=jsdom --watchAll=false

...

/usr/bin/node: bad option: --openssl-legacy-provider
```

Will remove it again in other targets, as it seems to run locally w/o that flag too.

**Comment by esron** [comment]
`--openssl-legacy-provider` is not supported by the pipeline image.

**Comment by esron** [comment]
This can be closed in favor of #198

**Pull request closed without merging**